resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -127,15 +127,18 @@ private[spark] class Client(
         .pods()
         .withName(driverPodName)
         .watch(watcher)) { _ =>
-      val createdDriverPod = kubernetesClient.pods().create(resolvedDriverPod)
+      var createdDriverPod: Option[Pod] = None
       try {
         val otherKubernetesResources =
           resolvedDriverSpec.driverKubernetesResources ++ Seq(configMap)
-        addDriverOwnerReference(createdDriverPod, otherKubernetesResources)
         kubernetesClient.resourceList(otherKubernetesResources: _*).createOrReplace()
+        createdDriverPod = Some(kubernetesClient.pods().create(resolvedDriverPod))
+        addDriverOwnerReference(createdDriverPod.get, otherKubernetesResources)
```

Comment:
This only adds the `OwnerReference` to the resource metadata to the local copies of the resources in memory, without updating the resources to the API server.

Reply:
Hi Yinan Li, thanks a lot for taking the look here. You are right here. But, I was wondering, can I do something to fix it. For example, if I do not create the driver pod before creating resources, the owner reference is not properly set up and if I do the other way, optional mount does not work(this issue).

Reply:
Basically you will need to update the resources against the API server after the driver pod gets created. However, there's a potential risk here: if the creation of the driver pod for whatever reasons, the ownerRefs won't be added so garbage collection won't kick in and consequently the resources won't be properly cleaned up.

Reply:
One way to avoid that risk is to have a [job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) and let the job own all the resources including the pod. What do you think?

Reply:
The idea of using a Job to run the driver has been rejected long time ago.

Reply:
@liyinan926 Thanks for helping me think through this, so far whatever solution I could think, I have no way to solve this problem without introducing another problem or risks. I will keep thinking, and if you get any hints, please help me with it. In the meantime, I am closing this PR.